doc/staging.md

```diff
@@ -0,0 +1,75 @@
+## Development workflow using staging.kernelci.org
+
+While the production instance is hosted on
+[kernelci.org](https://kernelci.org), another independent instance is hosted on
+[staging.kernelci.org](https://staging.kernelci.org).  This is where all the
+changes to the code and configuration get tested before updating production.
+It consists of Jenkins,
+[`kernelci-backend`](https://github.com/kernelci/kernelci-backend) for the
+Mongo DB and REST API, and
+[`kernelci-frontend`](https://github.com/kernelci/kernelci-frontend) for the
+web dashboard.
+
+Jobs are run every 8h on staging, using all the code from pull requests.  It
+will build a dedicated kernel branch based on
+[linux-next](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git)
+but with only a small number of configurations, and get some tests run in all
+the labs.
+
+### GitHub pull requests
+
+A special feature of the staging instance is the ability to test code from
+pending GitHub pull requests before they get merged.  This is handled by tools
+in the [`kernelci-deploy`](https://github.com/kernelci/kernelci-deploy)
+project, to pull all the open pull requests for a given project, apply some
+arbitrary patches and push the resulting `staging.kernelci.org` branch back to
+the repository.  This branch is being replaced (force-pushed) every time the
+tool is run.
+
+Things to note:
```

Comment:
Is it worth adding a note about the "staging-skip" label?

Reply:
Good point, I'll add that.  Thanks.

Reply:
Done, added a bullet point a bit further down.

Reply:
LGTM